beacon-chain/forkchoice/doubly-linked-tree/forkchoice.go

```diff
@@ -687,3 +705,18 @@ func (f *ForkChoice) Weight(root [32]byte) (uint64, error) {
 	}
 	return n.weight, nil
 }
+
+// updateJustifiedBalances updates the validators balances on the justified checkpoint pointed by root
+func (f *ForkChoice) updateJustifiedBalances(ctx context.Context, root [32]byte) error {
+	balances, err := f.balancesByRoot(ctx, root)
```

Comment:
Not clear to me if `checkpointsLock` should lock around balance updates too. If not, do we need a lock for it?

Reply:
yes, we do, I added a comment to this function, but I think I was careful to call it only when locked.